prettier/main.ts

```diff
@@ -227,6 +228,64 @@ async function formatSourceFiles(
   exit(0);
 }
 
+function filesSelector(
+  selectors: string[],
+  ignore: string[] = [],
+  options: GlobOptions = {}
+): AsyncIterableIterator<WalkInfo> {
+  const matchers: Array<string | RegExp> = [];
+
+  const selectorMap: { [k: string]: boolean } = {};
+
+  for (const selector of selectors) {
+    // if selector have exist. ignore it.
+    if (selectorMap[selector]) continue;
+    selectorMap[selector] = true;
+    if (isGlob(selector) || selector === ".") {
+      matchers.push(glob(selector, options));
+    } else {
+      matchers.push(selector);
+    }
+  }
+
+  const skip = ignore.map((i: string): RegExp => glob(i, options));
+
+  return (async function*(): AsyncIterableIterator<WalkInfo> {
+    for (const match of matchers) {
+      if (typeof match === "string") {
+        const fileInfo = await Deno.stat(match);
+
+        if (fileInfo.isDirectory()) {
+          const files = await Deno.readDir(match);
+          for (const f of files) {
+            const item: WalkInfo = {
+              filename: match + sep + f.name,
+              info: f
```

Comment:
If this `f` is again a directory, it seems a problem. How about using `walk` here as well?